src/question-bank/question-bank.service.ts

```diff
@@ -107,7 +115,64 @@ export class QuestionBankService {
     return this.prisma.questionBank.delete({
       where: {
         id: questionId,
+      },
+    });
+  }
+
+  async getAllQuestionsForUser(userId: string) {
+    const user = await this.mockUserService.findOne(userId);
+    if (!user) {
+      // Handle the case when the user is not found.
+      throw new Error("User not found");
+    }
+
+    // Get the user's designation
+    const designation = user.designation;
+
+    // Get all the roles for a designation
+    const userRoles =
+      await this.mockDesignationService.findAllRolesForDesignation(designation);
+
+    // Get roleId of each role of the user
+    const roleIds = userRoles?.Roles.map((item) => {
+      return item.id;
+    });
+
+    // unique competencyId associated with the roles of the user
+    const competencyIds: any = [];
+    if (roleIds?.length) {
+      for (let i = 0; i < roleIds.length; i++) {
+        const competency = (await this.mockRoleService.findRoleById(roleIds[i]))
+          .competencies;
+        competencyIds.push(
+          ...competency.filter((item) => {
+            if (!competencyIds.includes(item.competencyId)) {
+              competencyIds.push(item.competencyId);
+            }
+          })
+        );
+      }
+    }
+
+    // Get competencyLevel for an user
+    let competencyLevelNumber = user.Level?.levelNumber;
```

Comment:
User will contain only his highest level number or all levels he has?
If only the user contains the highest level number, are you fetching all below-level questions?

Reply:
Yes user will have highest level number and I am also fetching the question for lower levels also